pennylane/transforms/add_noise.py

```diff
@@ -18,17 +18,21 @@
 from pennylane.transforms.core import TransformContainer, transform
 
 
+# pylint: disable=too-many-branches
 @transform
 def add_noise(tape, noise_model, level=None):
     """Insert operations according to a provided noise model.
 
     Circuits passed through this quantum transform will be updated to apply the
     insertion-based :class:`~.NoiseModel`, which contains a mapping
     ``{BooleanFn: Callable}`` from conditions to the corresponding noise
-    gates. Each condition in the noise model will be evaluated on the
-    operations contained within the given circuit. For conditions that
-    evaluate to ``True``, the noisy gates contained within the ``Callable``
-    will be inserted after the operation under consideration.
+    gates for circuit operations and measurements each. First, each condition
+    in the first mapping of a noise model will be evaluated on the operations
```

Comment:
Is there a forced order for the mappings, i.e., the operator map should appear before the measurement map?

Reply:
By default, one would define the `model_map` before `meas_map` and the application happens in that order as well, which is reflected here.

Reply:
What happens if the user defines meas_map before model_map?

Reply:
They can define the maps in whichever order. However, when they are building a noise model, they are still forced to specify them in the required order due to the `__init__` signature, i.e., `qml.NoiseModel(model_map=model, meas_map=meas)`. 

The first argument (`model_map`) is a required one; it would be provided before the optional `meas_map`. Internally, by definition, we use the first map on the operations and the second one on the measurements for any given `tape`.